TelCo.ColorCoder/Program.cs

```diff
@@ -1,145 +1,40 @@
-            }
-
-            // Find the minor color in the array and get the index
-            int minorIndex = -1;
-            for (int i = 0; i < colorMapMinor.Length; i++)
-            {
-                if (colorMapMinor[i] == pair.minorColor)
-                {
-                    minorIndex = i;
-                    break;
-                }
-            }
-            // If colors can not be found throw an exception
-            if (majorIndex == -1 || minorIndex == -1)
-            {
-                throw new ArgumentException(
-                    string.Format("Unknown Colors: {0}", pair.ToString()));
-            }
-
-            // Compute pair number and Return  
-            // (Note: +1 in compute is because pair number is 1 based, not zero)
-            return (majorIndex * colorMapMinor.Length) + (minorIndex + 1);
-        }
-        /// <summary>
-        /// Test code for the class
-        /// </summary>
-        /// <param name="args"></param>
-        private static void Main(string[] args)
+        public static void Main(string[] args)
         {
             int pairNumber = 4;
-            ColorPair testPair1 = Program.GetColorFromPairNumber(pairNumber);
+            ColorPair testPair1 = GetColorFromPairNumber(pairNumber);
             Console.WriteLine("[In]Pair Number: {0},[Out] Colors: {1}\n", pairNumber, testPair1);
             Debug.Assert(testPair1.majorColor == Color.White);
             Debug.Assert(testPair1.minorColor == Color.Brown);
 
             pairNumber = 5;
-            testPair1 = Program.GetColorFromPairNumber(pairNumber);
+            testPair1 = GetColorFromPairNumber(pairNumber);
             Console.WriteLine("[In]Pair Number: {0},[Out] Colors: {1}\n", pairNumber, testPair1);
             Debug.Assert(testPair1.majorColor == Color.White);
             Debug.Assert(testPair1.minorColor == Color.SlateGray);
 
             pairNumber = 23;
-            testPair1 = Program.GetColorFromPairNumber(pairNumber);
+            testPair1 = GetColorFromPairNumber(pairNumber);
             Console.WriteLine("[In]Pair Number: {0},[Out] Colors: {1}\n", pairNumber, testPair1);
             Debug.Assert(testPair1.majorColor == Color.Violet);
             Debug.Assert(testPair1.minorColor == Color.Green);
 
             ColorPair testPair2 = new ColorPair() { majorColor = Color.Yellow, minorColor = Color.Green };
-            pairNumber = Program.GetPairNumberFromColor(testPair2);
+            pairNumber = GetPairNumberFromColor(testPair2);
             Console.WriteLine("[In]Colors: {0}, [Out] PairNumber: {1}\n", testPair2, pairNumber);
             Debug.Assert(pairNumber == 18);
 
             testPair2 = new ColorPair() { majorColor = Color.Red, minorColor = Color.Blue };
-            pairNumber = Program.GetPairNumberFromColor(testPair2);
+            pairNumber = GetPairNumberFromColor(testPair2);
             Console.WriteLine("[In]Colors: {0}, [Out] PairNumber: {1}", testPair2, pairNumber);
```

Comment:
follow the pattern, test the ToString() function by calling here.
Add Debug.Assert

Reply:
@madhurabindu  adding a test case for ToString() will increase the Line Of Code from more than 50, so can I remove one of the test case?

Reply:
No, Always leave the code you are working with in A better state than when you started. So you can not remove other test cases. You could however add more tests to existing tests if you want :)

So, Add the test for ToString() and think of how can the file be split logically. 

Hint: like you have Util class, you could have test class?

Reply:
@madhurabindu Please have a look now

Reply:
Good Job